src/main/java/com/openlattice/kindling/search/ConductorElasticsearchImpl.java

```diff
@@ -577,187 +546,59 @@ public boolean saveEntitySetToElasticsearch(
-        }
-        return false;
-    }
-
     /**
      * Add new mappings to existing index.
      * Updating the entity set model is handled in {@link #updatePropertyTypesInEntitySet(UUID, List)}
      *
-     * @param entitySetId        the id of the entity set to which the new proprties are added
-     * @param newPropertyTypes   the ids of the new properties
-     * @param linkedEntitySetIds if it's a linking entity set, the ids of the linked entity sets, otherwise empty
+     * @param entityType       the entity type to which the new properties are added
+     * @param newPropertyTypes the ids of the new properties
      */
     @Override
-    public boolean addPropertyTypesToEntitySet(
-            UUID entitySetId,
-            List<PropertyType> newPropertyTypes,
-            Optional<Set<UUID>> linkedEntitySetIds ) {
-        return addMappingToSecurableObjectIndex( entitySetId, newPropertyTypes, linkedEntitySetIds );
+    public boolean addPropertyTypesToEntityType( EntityType entityType, List<PropertyType> newPropertyTypes ) {
+        saveObjectToElasticsearch( ENTITY_TYPE_INDEX, ENTITY_TYPE, entityType, entityType.getId().toString() );
+        return addMappingToEntityTypeDataIndex( entityType, newPropertyTypes );
     }
 
     @Override
     public boolean addLinkedEntitySetsToEntitySet(
             UUID linkingEntitySetId,
             List<PropertyType> propertyTypes,
             Set<UUID> newLinkedEntitySets ) {
-        return addMappingToSecurableObjectIndex( linkingEntitySetId,
-                propertyTypes,
-                Optional.of( newLinkedEntitySets ) );
+        // TODO do we need this anymore?
+        return false;
+        //        return addMappingToEntityTypeDataIndex( linkingEntitySetId,
+        //                propertyTypes,
+        //                Optional.of( newLinkedEntitySets ) );
     }
 
     @Override
-    public boolean deleteEntitySet( UUID entitySetId ) {
+    public boolean deleteEntitySet( UUID entitySetId, UUID entityTypeId ) {
         if ( !verifyElasticsearchConnection() ) { return false; }
 
         client.prepareDelete( ENTITY_SET_DATA_MODEL, ENTITY_SET_TYPE, entitySetId.toString() ).execute().actionGet();
 
-        new DeleteByQueryRequestBuilder( client, DeleteByQueryAction.INSTANCE ).filter(
-                QueryBuilders.boolQuery()
-                        .must( QueryBuilders.matchQuery( TYPE_FIELD, ACLS ) )
-                        .must( QueryBuilders.matchQuery( ENTITY_SET_ID, entitySetId.toString() ) ) )
-                .source( ENTITY_SET_DATA_MODEL )
-                .execute()
-                .actionGet();
-
-        client.admin().indices()
-                .delete( new DeleteIndexRequest( getIndexName( entitySetId ) ) );
-
-        return true;
-    }
+        BulkByScrollResponse response = DeleteByQueryAction.INSTANCE.newRequestBuilder( client )
+                .filter( QueryBuilders.termQuery( ENTITY_SET_ID_KEY.toString(), entitySetId ) )
```

Comment:
We need to add linking entity set id to linked documents if we want to include them in query (see: SearchService. removeLinkedEntitySetsFromEntitySet)

Reply:
so in the new setup there should be one document per linked entity (globally, not one per linked entity set) so when a linking entity set is deleted, nothing needs to happen in elasticsearch. When a non-linking entity set (data entity set?) is deleted, this will delete all entities that are from that entity set. The linking entities won't be updated until the cluster is updated, although the deleted entity sets will be ignored in queries until that happens.

Reply:
Okay, I see. So there won't be an option to search by linking entity set id (since that was the index), only by including all of the normal entity sets it holds together and use all of them in the query. The one document part was clear, but I have to get used to the entity type centered indexing :)